src/Rings/puiseux_wrapper.jl

```diff
@@ -0,0 +1,92 @@
+########################################################################
+# Wrapper functionality for `PuiseuxExpansions.lib` in Singular        #
+########################################################################
+
+# A wrapper for Singular's `Singular.LibPuiseuxexpansions.puiseux`. 
+# It returns either a list of lists with entries of the following form: 
+#   * a polynomial `h`
+#   * an integer `e`
+#   * more stuff that is returned for internal purposes in the Singular library. 
+#     We discard this information
+# We interpret the result as `h` being the polynomial which evaluates 
+# to the Puiseux expansion of `f` on `y^(1//e)`. 
+#
+# Otherwise, it returns a ring and a dictionary with additional information 
+# on required extension fields for presentation of the result. Which one 
+# of the two cases applies is not foreseeable from the input alone. We 
+# therefore catch it with dispatch on `_process_result`. 
+
+function _process_result(P::MPolyRing, prec::Int, SE::Singular.PolyRing, res::Dict, rest...)
+  # create the necesessary field extension on the Oscar side
+  kk = coefficient_ring(SE) # the extension field of QQ
+  mp = Singular.modulus(kk) # the minimum polynomial
+  mmp = Singular.n_transExt_to_spoly(mp) # convert into an actual polynomial
+  L, t = polynomial_ring(QQ, :t; cached=false)
+  kkO, alpha = extension_field(L(mmp))
+  P_ext, to_P_ext = change_base_ring(kkO, P)
+  Puis, xx = puiseux_series_ring(kkO, prec, symbols(P_ext)[1])
```

Comment:
```suggestion
  Puis, xx = puiseux_series_ring(kkO, prec, symbols(P_ext)[1]; cached=false)
```
library code should not create cached parents